auth/presentation/src/main/java/com/nibalk/tasky/auth/presentation/register/RegisterScreen.kt

```diff
@@ -84,12 +98,27 @@ private fun RegisterScreen(
                 onAction(RegisterAction.OnTogglePasswordVisibilityClick)
             },
             hint = stringResource(id = R.string.auth_password),
-            modifier = Modifier.fillMaxWidth()
-        )
-        // Password Requirements
-        Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceMedium))
-        PasswordRequirements(
-            passwordValidationState = state.passwordValidationState
+            modifier = Modifier.fillMaxWidth(),
+            error = if(!state.passwordValidationState.hasMinLength) {
+                stringResource(
+                    id = R.string.auth_at_least_x_characters,
+                    AuthDataValidator.PASSWORD_MIN_LENGTH
+                )
+            } else if (!state.passwordValidationState.hasNumericCharacter) {
+                stringResource(
+                    id = R.string.auth_at_least_one_number,
+                )
+            } else if (!state.passwordValidationState.hasLowerCaseCharacter) {
+                stringResource(
+                    id = R.string.auth_at_least_one_lowercase_char,
+                )
+            } else if (!state.passwordValidationState.hasUpperCaseCharacter) {
+                stringResource(
+                    id = R.string.auth_at_least_one_uppercase_char,
+                )
```

Comment:
IMO, for such long else-ifs, `when` expressions usually become more readable

Reply:
Thanks for all your feedback. Will adjust everything from my next PR :)